### PR TITLE
Handle errors when adding repo

### PR DESCRIPTION
**Pull request by VoxSecundus**
This PR adds some extra handling to the `repo add` command to halt the process if there was an error interacting with the cloud provider. It also changes the `Choose a provider` interface to be a select menu with options for each of the existing providers.

### CODE REVIEW

File: lib/silo/commands/repo_add.rb [hunk 1]
```diff
@@ -65,7 +75,7 @@ def run
         puts "Silo added"
       end
 
-      def prompt
+     def prompt
```
Comment (Womblue) [inline lib/silo/commands/repo_add.rb:78]: Assuming this is a typo.

**Review by Womblue — CHANGES_REQUESTED** [review]
Minor indentation typo, up to you whether to merge this now or save it for the branch that changes all commands to use `open3`

File: lib/silo/commands/repo_add.rb (hunk 1 above)
Comment (VoxSecundus) [reply lib/silo/commands/repo_add.rb:78]: Typo, indeed.

**Comment by VoxSecundus** [comment]
I plan on getting to the Open3 changes tonight, so I'll leave this open for now and rebase it in the morning.

**Comment by VoxSecundus** [comment]
> I plan on getting to the Open3 changes tonight, so I'll leave this open for now and rebase it in the morning.

Changed my mind; it's a bug that ought to be merged because it exists in the https://github.com/openflighthpc/flight-silo/pull/13 branch.

**Pull request merged**